core/src/main/resources/org/apache/spark/ui/static/historypage.js

```diff
@@ -54,7 +54,8 @@ function makeIdNumeric(id) {
 }
 
 function formatDate(date) {
-  return date.split(".")[0].replace("T", " ");
+  if (date <= 0) return "-";
```

Comment:
I personally think that just adds more noise to the screen. Sometimes the column value is a date, sometimes a text string then. Using a blank is the most natural "no value" to me.